src/components/Nabla/useSwapForm.tsx

```diff
@@ -22,17 +22,25 @@ export const useSwapForm = () => {
   const setTokenModal = tokensModal[1];
 
   const initialState = useMemo(() => {
-    const storageValues = storageService.getParsed<SwapSettings>(storageKeys.SWAP_SETTINGS);
+    const searchParams = new URLSearchParams(window.location.search);
 
-    const storedFromTokenIsValid = INPUT_TOKEN_CONFIG[storageValues?.from as InputTokenType] !== undefined;
-    const storedToTokenIsValid = OUTPUT_TOKEN_CONFIG[storageValues?.to as OutputTokenType] !== undefined;
+    const defaultValues = { from: 'usdc', to: 'eurc' };
+    const storageValues = storageService.getParsed<SwapSettings>(storageKeys.SWAP_SETTINGS);
+    const searchParamValues = { from: searchParams.get('from'), to: searchParams.get('to') };
 
-    return {
-      from: (storedFromTokenIsValid ? storageValues?.from : 'usdc') as InputTokenType,
-      to: (storedToTokenIsValid ? storageValues?.to : 'eurc') as OutputTokenType,
-      taxNumber: '',
-      bankAccount: '',
+    const initialValues = {
+      ...defaultValues,
+      ...searchParamValues,
+      ...storageValues,
```

Comment:
Shouldn't `searchParamValues` have higher priority than those stored? In case of a redirect.

Reply:
I agree, changed.